spotify_player/src/client/mod.rs

```diff
@@ -1383,22 +1381,23 @@ impl Client {
         ))
         .replace('/', ""); // remove invalid characters from the file's name
         let path = configs.cache_folder.join("image").join(path);
-
-        #[cfg(feature = "image")]
-        if !state.data.read().caches.images.contains_key(url) {
-            let bytes = self
-                .retrieve_image(url, &path, configs.app_config.enable_cover_image_cache)
-                .await?;
-            let image =
-                image::load_from_memory(&bytes).context("Failed to load image from memory")?;
-            state
-                .data
-                .write()
-                .caches
-                .images
-                .insert(url.to_owned(), image, *TTL_CACHE_DURATION);
+        if configs.app_config.enable_cover_image_cache{
+            if !state.data.read().caches.images.contains_key(url) {
+                let bytes = self
+                    .retrieve_image(url, &path, configs.app_config.enable_cover_image_cache)
+                    .await?;
+                let image =
+                    image::load_from_memory(&bytes).context("Failed to load image from memory")?;
+                state
+                    .data
+                    .write()
+                    .caches
+                    .images
+                    .insert(url.to_owned(), image, *TTL_CACHE_DURATION);
+            }
         }
-
+        
+        #[cfg(any(feature = "image", feature = "notify"))]
```

Comment:
you can keep `MemoryCaches` and `image` feature optional by updating the logic:

```suggestion
        if configs.app_config.enable_cover_image_cache{
            self
                .retrieve_image(url, &path, true)
                .await?;
        }

        #[cfg(feature = "image")]
        if !state.data.read().caches.images.contains_key(url) {
            let bytes = self
                .retrieve_image(url, &path, false)
                .await?;
            let image =
                image::load_from_memory(&bytes).context("Failed to load image from memory")?;
            state
                .data
                .write()
                .caches
                .images
                .insert(url.to_owned(), image, *TTL_CACHE_DURATION);
        }
```

Reply:
Ahh yeah, that makes sense; I don't know why I was trying to avoid adding that extra code block. I'll add those checks back in and recommit. Thanks!

Reply:
Made the suggested changes and removed the loose Feature check since it was no longer needed.

Reply:
can you undo the change in `Cargo.toml` as well?

Reply:
Oh whoops, absolutely. Re-pushed.